### PR TITLE
CI jobs updated + some linting

### PR DESCRIPTION
**Pull request by juleskers**
Since my review comments crossed @pserwylo 's merge, here is an updated actions-yaml with my suggestions.

Since I made the mistake of opening the lint-report from the output, some linter-cleanups are thrown in as well :sweat_smile: 

This has run through successfully: https://github.com/nicolasbrailo/PianOli/actions/runs/6267155391/job/17019621693

### CODE REVIEW

**Comment by juleskers** [comment]
HA! and because I've now opened a PR, it's running _again_, even though nothing changed...

I guess it's redundant to build for "PR against master" if we already build "branches=*" anyway?

**Comment by juleskers** [comment]
@pserwylo: I think you should definitely have a look at at least the gradle-validation step, with an eye for porting it to all of your other projects.
https://github.com/gradle/wrapper-validation-action#readme  

Supply chain attacks against ones personal laptop are no joke!

File: .github/workflows/submit-dependency-snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Submits dependency graph info generated by other jobs to GitHub API
+# This requires content-write permission, which is by default disabled if the PR-branch is from a "foreign" repo,
+# https://github.com/gradle/gradle-build-action#enable-dependency-graph-generation-for-a-workflow
+
+name: submit-dependency-snapshot
+
+on:
+  workflow_run:
+    workflows: ['Build and test app']
```
Comment (juleskers) [inline .github/workflows/submit-dependency-snapshot.yml:9]: Note for review: this won't trigger until this file is on the default branch.

See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run

**Comment by pserwylo** [comment]
> @pserwylo: I think you should definitely have a look at at least the gradle-validation step, with an eye for porting it to all of your other projects. https://github.com/gradle/wrapper-validation-action#readme
> 
> Supply chain attacks against ones personal laptop are no joke!

Sounds sensible. Random note though, most of the time we trust F-Droid, which has its own Gradle setup, and I trust the team much more than my own laptop. For apps built before August 2021, I publish the F-Droid build directly to Google play (Lexica, Super retro mega wars, baby dots, some other apps of mine). That signature is what gets delivered to end users of Google Play.

For more recent apps of mine, I still upload the F-Droid build to Google Play, but they then strip that signature and add their own. Alas, for PianOli, I need to build on my laptop because someone took our package id!

**Comment by juleskers** [comment]
> Random note though, most of the time we trust F-Droid, which has its own Gradle setup, and I trust the team much more than my own laptop.

The point of that validation is, the wrapper gets executed, with the user permission, not just for builds, but even each time your IDE refreshes its settings (git checkout of new branch? Wrapper runs. Open project? Wrapper runs. Change a typo in `build.gradle`? Wrapper runs. Etc.)

It's a very attractive target to add malware hooks to. Not just "send all ssh keys and signing keys to criminals"-type, but even stupid stuff like "start a crypto-miner in the background", or "join a botnet as anonymising relay node".

**Pull request merged**